src/Microsoft.Framework.FileSystemGlobbing/Matcher.cs

```diff
@@ -10,24 +10,24 @@ namespace Microsoft.Framework.FileSystemGlobbing
 {
     public class Matcher
     {
-        private IList<IPattern> _includePatterns = new List<IPattern>();
-        private IList<IPattern> _excludePatterns = new List<IPattern>();
+        protected IList<IPattern> IncludePatterns { get; set; } = new List<IPattern>();
+        protected IList<IPattern> ExcludePatterns { get; set; } = new List<IPattern>();
```

Comment:
I wondered whether the properties shouldn't just be IEnumerable to enable inspection, and leave the Add/Remove as distinct APIs

-----Original Message-----
From: "Eilon Lipton" <notifications@github.com>
Sent: ‎2/‎16/‎2015 2:32 PM
To: "aspnet/FileSystem" <FileSystem@noreply.github.com>
Cc: "Damian Edwards" <damian@damianedwards.com>
Subject: Re: [FileSystem] Make Matcher more testable (#63)

In src/Microsoft.Framework.FileSystemGlobbing/Matcher.cs:
> @@ -10,24 +10,24 @@ namespace Microsoft.Framework.FileSystemGlobbing
>  {
>      public class Matcher
>      {
> -        private IList<IPattern> _includePatterns = new List<IPattern>();
> -        private IList<IPattern> _excludePatterns = new List<IPattern>();
> +        protected IList<IPattern> IncludePatterns { get; set; } = new List<IPattern>();
> +        protected IList<IPattern> ExcludePatterns { get; set; } = new List<IPattern>();
How do you guard against a derived class from adding invalid patterns? Changing a class's design just for a unit test is a slippery slope.
—
Reply to this email directly or view it on GitHub.